File: CS/SaveImageToDB/Controllers/HomeController.cs
using System;
using System.IO;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.ASPxUploadControl;
using DevExpress.Web.Mvc;

namespace SaveImageToDB.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult HtmlEditorPartial() {
            return PartialView();
        }

        public ActionResult ImageUpload() {
            HtmlEditorDemosHelper.SaveUploadedImage(HtmlEditorDemosHelper.ImageUploadValidationSettings);
            return null;
        }
    }

    public class HtmlEditorDemosHelper {
        public const string ImagesDirectory = "~/Content/Images/";
        public const string ThumbnailsDirectory = "~/Content/Thumbnails/";
        public const string UploadDirectory = ImagesDirectory + "Upload/";

        public static void SaveUploadedImage(ValidationSettings validationSettings) {
            EventHandler<FileUploadCompleteEventArgs> fileUploadComplete = (object sender, FileUploadCompleteEventArgs e) => {
                DBUtils.SaveImageToDB(e.UploadedFile.FileBytes);

                string fileName = e.UploadedFile.FileName;
                string filePath = HttpContext.Current.Request.MapPath(Path.Combine(UploadDirectory, fileName));
                e.UploadedFile.SaveAs(filePath);
                e.CallbackData = fileName;
            };
            UploadControlExtension.GetUploadedFiles(GetUploadControlID(), validationSettings, fileUploadComplete);
        }

        protected static string GetUploadControlID() {
            string helperParamName = DevExpress.Web.ASPxClasses.Internal.RenderUtils.HelperUploadingCallbackQueryParamName;
            string uploadParamName = DevExpress.Web.ASPxClasses.Internal.RenderUtils.UploadingCallbackQueryParamName;
            
            return HttpContext.Current.Request.Params[helperParamName] ?? HttpContext.Current.Request.Params[uploadParamName];
        }

        public static readonly ValidationSettings ImageUploadValidationSettings = new ValidationSettings {
            AllowedFileExtensions = new string[] { ".jpg", ".jpeg", ".jpe", ".gif" },
            MaxFileSize = 4000000
        };
    }
}

File: VB/SaveImageToDB/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpress.Web.ASPxFileManager
Imports DevExpress.Web.ASPxHtmlEditor
Imports DevExpress.Web.ASPxUploadControl
Imports System.Web.UI.WebControls
Imports DevExpress.Utils
Imports System.IO

Namespace SaveImageToDB.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function HtmlEditorPartial() As ActionResult
            Return PartialView("_PartialHtmlEditor")
        End Function

        Public Function ImageUpload() As ActionResult
            HtmlEditorDemosHelper.SaveUploadedImage(HtmlEditorDemosHelper.ImageUploadValidationSettings)
            Return Nothing
        End Function
    End Class

    Public Class HtmlEditorDemosHelper
        Public Const ImagesDirectory As String = "~/Content/Images/"
        Public Const ThumbnailsDirectory As String = "~/Content/Thumbnails/"
        Public Const UploadDirectory As String = ImagesDirectory & "Upload/"

        Public Shared Sub SaveUploadedImage(ByVal validationSettings As ValidationSettings)
            Dim fileUploadComplete As EventHandler(Of FileUploadCompleteEventArgs) = Sub(sender As Object, e As FileUploadCompleteEventArgs)
                DBUtils.SaveImageToDB(e.UploadedFile.FileBytes)

                Dim fileName As String = e.UploadedFile.FileName
                Dim filePath As String = HttpContext.Current.Request.MapPath(Path.Combine(UploadDirectory, fileName))
                e.UploadedFile.SaveAs(filePath)
                e.CallbackData = fileName
            End Sub
            UploadControlExtension.GetUploadedFiles(GetUploadControlID(), validationSettings, fileUploadComplete)
        End Sub
        Protected Shared Function GetUploadControlID() As String
            Return HttpContext.Current.Request.Params(DevExpress.Web.ASPxClasses.Internal.RenderUtils.UploadingCallbackQueryParamName)
        End Function

        Public Shared ReadOnly ImageUploadValidationSettings As ValidationSettings = New ValidationSettings With {.AllowedFileExtensions = New String() { ".jpg", ".jpeg", ".jpe", ".gif" }, .MaxFileSize = 4000000}
    End Class
End Namespace
